lumicks/pylake/population/dwelltime.py

```diff
@@ -800,13 +848,17 @@ def _exponential_mixture_log_likelihood_jacobian(params, t, t_min, t_max):
     t = t[np.newaxis, :]
 
     exp_bound_difference = np.exp(-t_min / lifetimes) - np.exp(-t_max / lifetimes)
-    norm_factor = 1.0 / np.sum(amplitudes * exp_bound_difference)
+    norm_factor = 1.0 / np.sum(amplitudes * exp_bound_difference, axis=0)
 
     # Calculate derivatives of the normalization constant
     dnorm_damp = -(norm_factor**2) * exp_bound_difference
     dlognorm_damp = (1.0 / norm_factor) * dnorm_damp
 
-    max_bound = t_max * np.exp(-t_max / lifetimes) if np.all(t_max / lifetimes < 1e10) else 0
+    max_div_lifetimes = t_max / lifetimes
+    valid = max_div_lifetimes < 1e10
+    max_bound = np.zeros(max_div_lifetimes.shape)
+    max_bound[valid] = (t_max * np.ones(lifetimes.shape))[valid] * np.exp(-max_div_lifetimes[valid])
```

Comment:
could save on some parentheses to make this a bit cleaner
```suggestion
    max_bound[valid] = np.full(lifetimes.shape, t_max)[valid] * np.exp(-max_div_lifetimes[valid])
```

Reply:
That's not the same though. `t_max` can either be an array or scalar.

Reply:
oh yeah that's the whole point of this PR 😖